vcr/oidc4vci/validators.go

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2023 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package oidc4vci
+
+import (
+	"errors"
+	ssi "github.com/nuts-foundation/go-did"
+	"github.com/nuts-foundation/go-did/vc"
+)
+
+// ValidateCredentialDefinition validates the CredentialDefinition according to the VerifiableCredentialJSONLDFormat format
+func ValidateCredentialDefinition(definition *CredentialDefinition, isOffer bool) error {
+	if definition == nil {
+		return errors.New("invalid credential_definition: missing")
+	}
+	if len(definition.Context) == 0 {
+		return errors.New("invalid credential_definition: missing @context field")
+	}
+	if len(definition.Type) == 0 {
+		return errors.New("invalid credential_definition: missing type field")
+	}
+	if definition.CredentialSubject != nil {
+		if isOffer {
+			return errors.New("invalid credential_definition: credentialSubject not allowed in offer")
+		}
+		// TODO: add credentialSubject validation.
```

Comment:
a todo

Reply:
we currently do not use the credentialSubject, and I am not sure what to validate here. We could add some basic validation but that might give us more issues later on if the validation turns out to be incorrect.

Reply:
then it's not a todo for the code. If we need to remember something, it should be listed in an issue not in the code.

Reply:
created issue #2320